src/AppInstallerCommonCore/Filesystem.cpp

```diff
@@ -238,4 +238,18 @@ namespace AppInstaller::Filesystem
         THROW_IF_FAILED(SHGetKnownFolderPath(id, KF_FLAG_NO_ALIAS | KF_FLAG_DONT_VERIFY | KF_FLAG_NO_PACKAGE_REDIRECTION, NULL, &knownFolder));
         return knownFolder.get();
     }
+
+    bool IsSameVolume(const std::filesystem::path& path1, const std::filesystem::path& path2)
+    {
+        std::wstring path1Volume;
+        std::wstring path2Volume;
+        DWORD path1Length = static_cast<DWORD>(path1.u16string().length());

```

Comment:
>DWORD path1Length = static_cast<DWORD>(path1.u16string().length()); [](http://example.com/codeflow?start=8&length=67)

DWORD path1VolumeLength = static_cast<DWORD>(path1.wstring().length() + 1);

Same for below